patches/chrome-browser-ui-BUILD.gn.patch

```diff
@@ -10,7 +10,15 @@ index 768e11b9e56483cbe6235c894bca669b928e4d0c..ba693710323ce67fa923ae9b66346a04
  
    if (enable_vr && is_win) {
      deps += [ "//chrome/browser/vr:vr_base" ]
-@@ -4606,7 +4607,10 @@ static_library("ui") {
+@@ -4477,6 +4478,7 @@ static_library("ui") {
+       "webui/settings/settings_utils_mac.mm",
+       "window_sizer/window_sizer_mac.mm",
+     ]
++    import("//brave/updater/config.gni") if (!brave_enable_updater) { sources -= [ "webui/help/version_updater_mac.mm" ] sources += [ "//brave/chromium_src/chrome/browser/ui/webui/help/version_updater_mac_sparkle.mm" ] }
```

Comment:
How about this instead of patching and renaming?
```
diff --git a/chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm b/chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm
index 88bd1929e85..bb8210e6016 100644
--- a/chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm
+++ b/chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm
@@ -3,6 +3,11 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */

+#include "brave/updater/buildflags.h"
+
+#if BUILDFLAG(BRAVE_ENABLE_UPDATER)
+#include "src/chrome/browser/ui/webui/help/version_updater_mac.mm"
+#else
 #include "brave/chromium_src/chrome/browser/ui/webui/help/version_updater_mac.h"

 #include <memory>
@@ -207,3 +212,4 @@ void VersionUpdaterMac::UpdateStatus(NSDictionary* dictionary) {
 void VersionUpdaterMac::UpdateShowPromoteButton() {
   NOTIMPLEMENTED();
 }
+#endif
```

Reply:
This does seem more elegant than what I have. I wonder if it will rebuild when `src/chrome/browser/ui/webui/help/version_updater_mac.mm` is touched. I'll try it! Thanks :)

Reply:
It worked!

Reply:
Great! and need deps for this like below as we're including new header :)  

```
diff --git a/chromium_src/chrome/browser/ui/BUILD.gn b/chromium_src/chrome/browser/ui/BUILD.gn
index f33c87c0c66..141cd474d9e 100644
--- a/chromium_src/chrome/browser/ui/BUILD.gn
+++ b/chromium_src/chrome/browser/ui/BUILD.gn
@@ -11,8 +11,13 @@ group("ui") {
   deps = [
     "//brave/components/brave_player/common/buildflags",
     "//brave/components/brave_vpn/common/buildflags",
+    "//brave/updater:buildflags",
   ]

+  if (is_mac) {
+    deps += [ "//brave/updater:buildflags" ]
+  }
+
   if (toolkit_views) {
     deps += [
       "//brave/components/ai_chat/core/common/buildflags",
```

Reply:
Fixed in ebda2fff76801f6732942c74c7f33d029a52465a.